src/main/java/me/shivzee/JMailTM.java

```diff
@@ -119,20 +119,22 @@ private Message messageUtility(JSONObject json) throws ParseException, DateTimeP
         boolean hasAttachments = (boolean) json.get("hasAttachments");
 
         List<Attachment> attachments = new ArrayList<>();
-        JSONArray attachmentArray = (JSONArray) parser.parse(json.get("attachments").toString());
-        Object [] aArray = attachmentArray.toArray();
-
-        for(Object attachmentObject : aArray){
-            JSONObject object = (JSONObject) attachmentObject;
-            String aId = object.get("id").toString();
-            String aFilename = object.get("filename").toString();
-            String aContentType = object.get("contentType").toString();
-            String aDisposition = object.get("disposition").toString();
-            String aTransferEncoding = object.get("transferEncoding").toString();
-            boolean aRelated = (boolean) object.get("related");
-            long aSize = Long.parseLong(object.get("size").toString());
-            String aDownloadUrl = object.get("downloadUrl").toString();
-            attachments.add(new Attachment(aId , aFilename , aContentType , aDisposition, aTransferEncoding ,aRelated ,aSize , aDownloadUrl ,bearerToken));
+        if(hasAttachments) {

```

Comment:
```java
 if(hasAttachments) {
            JSONArray attachmentArray = (JSONArray) parser.parse(json.get("attachments").toString());
            Object[] aArray = attachmentArray.toArray();

            for (Object attachmentObject : aArray) {
                JSONObject object = (JSONObject) attachmentObject;
                 ... 
                attachments.add(new Attachment(aId, aFilename, aContentType, aDisposition, aTransferEncoding, aRelated, aSize, aDownloadUrl, bearerToken));
            }
        }
  ```
  
  here it would be better if you do it like this .. instead of wrapping it in whole `if` statement you can have the array as empty or filled.
it would not loop with 0 elements and will make the code more readable
  
  ```java
  
            JSONArray attachmentArray = (JSONArray) parser.parse(json.get("attachments").toString());
            Object[] aArray = new Object[0];
            if(hasAttachments) {
                   aArray = attachmentArray.toArray();
            }
            
            for (Object attachmentObject : aArray) {
                JSONObject object = (JSONObject) attachmentObject;
                 ... 
                attachments.add(new Attachment(aId, aFilename, aContentType, aDisposition, aTransferEncoding, aRelated, aSize, aDownloadUrl, bearerToken));
            }
```